canonical-web-and-design/maas-ui
# feat(ui): add ability to generate client certificate when adding LXD VM host

## Done

- Added `generateCertificate` actions/reducers/selectors
- Dispatch action to generate certificate in the Add LXD form and display in the authentication step

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and click "Add KVM"
- Choose LXD, then input a name and address and make sure "Generate certificate" is selected
- Check that actions are dispatched to generate a certificate, and once loaded it shows up in the next step of the form

## Fixes

Fixes canonical-web-and-design/app-squad#257
